backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -132,4 +142,109 @@ export class IdimWebserviceService {
+            // set xml schema parameters for the request
+            const requestData = {
+                accountDetailRequest: {
+                    onlineServiceId: this.idimWebServiceID,
+                    // who is sending the request
+                    requesterAccountTypeCode,
+                    requesterUserGuid,
+                    // who we search for, exact match userID
+                    userId,
+                    accountTypeCode: RequesterAccountTypeCode.Business,
+                },
+            };
+
+            return new Promise((resolve, reject) => {
+                client.BCeIDService.BCeIDServiceSoap.getAccountDetail(
+                    requestData,
+                    function (error, foundUser) {
+                        // this will be any error from the IDIM server side
+                        if (error) {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+                        }
+
+                        if (
+                            foundUser.getAccountDetailResult.code == 'Failed' &&
+                            foundUser.getAccountDetailResult.failureCode ==
+                                'NoResults'
+                        ) {
+                            const response = new BCEIDUserResponse();
+                            response.found = false;
+                            response.userId = userId;
+                            return resolve(response);
+                        }
+
+                        // this will be any error return by the web service
+                        // for example if we provided an non existing requestor id, or permission issue
+                        if (foundUser.getAccountDetailResult.code == 'Failed') {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        status: HttpStatus.BAD_REQUEST,
+                                        code: foundUser.getAccountDetailResult
+                                            .code,
+                                        failureCode:
+                                            foundUser.getAccountDetailResult
+                                                .failureCode,
+                                        message:
+                                            foundUser.getAccountDetailResult
+                                                .message,
+                                    },
+                                    HttpStatus.BAD_REQUEST
```

Comment:
Hi Catherine,  wondering if this code is the same as line 102 for intended HttpException?